raiden/transfer/mediated_transfer/target.py

```diff
@@ -68,8 +68,10 @@ def events_for_onchain_secretreveal(
         channel_state.partner_state,
         transfer.lock.secrethash,
     )
+    has_onchain_reveal_started = target_state.state == 'onchain_reveal_secret'
 
-    if not safe_to_wait and secret_known_offchain:
+    if not safe_to_wait and secret_known_offchain and not has_onchain_reveal_started:
+        target_state.state = 'onchain_reveal_secret'
```

Comment:
This should go:
https://github.com/raiden-network/raiden/blob/6782cc536c29f9bebbb9574fde43e425515c2795/raiden/transfer/mediated_transfer/state.py#L286-L290